Improve documentation of sendTransaction (#14770)
solana-labs/solana: docs/src/developing/clients/jsonrpc-api.md

- Hash: A SHA-256 hash of a chunk of data.
- Pubkey: The public key of a Ed25519 key-pair.

> - Transaction: A list of Solana instructions signed by a client key-pair to authorize those actions.

Reviewer: ```suggestion
- Transaction: A list of Solana instructions signed by a client keypair to authorize those actions.
```
Nit: I know this predates your changes, but this is the only place we hyphenate keypair...

Author: oh, consistency outweighs. :)

The JSON structure of a transaction is defined as follows:

> - `signatures: <array[string]>` - A list of base-58 encoded signatures applied to the transaction. The list is always of length `message.header.numRequiredSignatures` and not empty. The signature at index `i` corresponds to the public key at index `i` in `message.account_keys`. The first one is used as [transaction id](../../terminology.md#transaction-id).

Reviewer: ```suggestion
- `signatures: <array[string]>` - A list of base-58 encoded signatures applied to the transaction. The list is always of length `message.header.numRequiredSignatures` and not empty. The signature at index `i` corresponds to the public key at index `i` in `message.account_keys`. The first one is used as the [transaction id](../../terminology.md#transaction-id).
```

Submits a signed transaction to the cluster for processing.

> Note that this doesn't guarantee the transaction is confirmed by the cluster.
> While this method reasonably retries to submit it background, the transaction
> could be rejected if transaction's `recent_blockhash` expires.
> 
> In other words, this method immediately succeeds, without waiting for any
> confirmations.
> 
> Use [`getSignatureStatuses`](jsonrpc-api.md#getsignaturestatuses) to make sure
> the transaction is confirmed or not.
> 
> Also, this method doesn't alter the transaction in any way. This method just
> relays the transaction created by clients to the cluster as-is.

Reviewer: ```suggestion
This method does not alter the transaction in any way; it relays the transaction created by clients to the node as-is.

If the node's rpc service receives the transaction, this method immediately succeeds, without waiting for any confirmations. A successful response from this method does not guarantee the transaction is processed or confirmed by the cluster.

While the rpc service will reasonably retry to submit it, the transaction
could be rejected if transaction's `recent_blockhash` expires before it lands.

Use [`getSignatureStatuses`](jsonrpc-api.md#getsignaturestatuses) to ensure
a transaction is processed and confirmed.
```

Reviewer: (Sorry for the lack of proper wrapping... github doesn't make that easy in suggestions :( )

   disabled if desired. It is recommended to specify the same commitment and
   preflight commitment to avoid confusing behavior.

> Also, the returned signature isn't something this method specifically creates or
> registers at the cluster. It is just the first signature in the transaction, which
> can be used to identify the transaction by other methods.
> In fact, the signature can easily extracted from the transaction data.

Reviewer: ```suggestion
The returned signature is the first signature in the transaction, which
is used to identify the transaction. This identifier can be easily extracted from the transaction data before submission.
```

Reviewer: Plus maybe link to a definition in terminology, if you decide to go that route?

#### Results:

> - `<string>` - First Transaction Signature embedded in the transaction, as base-58 encoded string (to be used for identification of transaction)

Reviewer: Maybe it would be better to define something like "transaction ID" in https://docs.solana.com/terminology and link to it from here (and other places)?

Author: done!
